diagnostic.md

```diff
@@ -3,36 +3,40 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+1. Organizing code into modules.
+2. Setting up build tools.
+3. Client side rendering.
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
 ```sh
-# your answer here
+The templating system in Ember is used by the router to display relevant information to the user in the view-state. Each template was written in Handlebars, and could access and manipulate properties in the View. In-built Handlebars helpers such as {{#if}} and {{#each}} were also available.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+Data layer, model layer.
```

Comment:
The layers we talked about were:
Router
Routes
Models
Templates
Components
Adapters
Controllers (1.0)
Views(1.0)

.5/1